crates/cext/src/transpiler/target.rs

```diff
@@ -1253,6 +1254,223 @@ pub struct CInstructionProperties {
     pub error: f64,
 }
 
+/// Representation of an operation identified within the Target.
+#[repr(C)]
+pub struct CTargetOperation {
+    /// The identifier for the current operation.
+    pub op_type: COperationKind,
+    /// The name of the operation.
+    pub name: *mut c_char,
+    /// The number of qubits this operation supports.
+    pub num_qubits: u32,
+    /// The parameters tied to this operation if fixed, as a `double`.
+    pub params: *mut f64,
+    /// The number of parameters supported by this operation.
+    pub num_params: u32,
+}
+
+/// @ingroup QkTarget
+/// Retrieves information about an operation in the Target.
+///
+/// @param target A pointer to the ``QkTarget``.
+/// @param index The index in which the gate is stored.
+/// @param op_kind A pointer to the space where the ``QkTargetOperation``
+///     will be stored.
+///
+/// @return An exit code.
+///
+/// # Example
+/// ```c
+///     QkTarget *target = qk_target_new(5);
+///
+///     QkTargetEntry *entry = qk_target_entry_new(QkGate_CX);
+///     uint32_t qargs[2] = {0, 1};
+///     qk_target_entry_add_property(entry, qargs, 2, 0.0, 0.1);
+///     qk_target_add_instruction(target, entry);
+///
+///     QkTargetOperation op;
+///     qk_target_op_get(target, 0, &op);
+///     
+///     // Clean up after you're done
+///     qk_target_op_clear(&op);
+/// ```
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+/// Behavior is undefined if ``inst_props`` does not point to an address of the correct size to
+/// store ``QkInstructionProperties`` in.
+#[unsafe(no_mangle)]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_op_get(
+    target: *const Target,
+    index: usize,
+    op_kind: *mut CTargetOperation,
+) -> ExitCode {
```

Comment:
I'm fine with `op_get` here.  `op_info`'s another choice if you prefer to avoid `_get`.

If I'm reading this correctly, the only error return is the index, which I think we've agreed in principle to just treat as a panic in the C API, at least until we're in a position to add some sort of "checked" API in addition.  We can probably just remove the `ExitCode` return here, I think?